### PR TITLE
Genprint v0.2

### PR DESCRIPTION
**Pull request by progman1**
library for printing of values vis a vis the ocaml interpreter, but from within compilation units.

### CODE REVIEW

File: packages/genprint/genprint.0.2/opam [hunk 1]
```diff
@@ -1,6 +1,6 @@
 opam-version: "2.0"
 #name: "genprint"
```
Comment (dra27) [inline packages/genprint/genprint.0.2/opam]:
```suggestion
```

File: packages/genprint/genprint.0.2/opam [hunk 2]
```diff
@@ -1,6 +1,6 @@
...
-#version: "0.1"
+#version: "0.2"
```
Comment (dra27) [inline packages/genprint/genprint.0.2/opam]: These commented out lines serve no purpose - why not just delete them?

File: packages/genprint/genprint.0.2/opam [hunk 3]
```diff
@@ -16,19 +16,18 @@ dev-repo: "git+https://github.com/progman1/genprintlib.git"
...
 ]
 
 depends: [
   "ocaml" {>= "4.02.0" }
   "dune"
-  "stdlib-shims"
-  "ppxlib" {build}
+  "ppxlib"
+  "cppo"
+  "stdlib-shims" {build}
```
Comment (dra27) [inline packages/genprint/genprint.0.2/opam]:
```suggestion
  "stdlib-shims"
```
`stdlib-shims` should not be tagged build - if it's updated, your package needs to rebuild too

File: packages/genprint/genprint.0.2/opam [hunk 4]
```diff
@@ -16,19 +16,18 @@ dev-repo: "git+https://github.com/progman1/genprintlib.git"
...
   ["dune" "build" "-p" name "-j" jobs]
 ]
 
 depends: [
   "ocaml" {>= "4.02.0" }
   "dune"
-  "stdlib-shims"
-  "ppxlib" {build}
+  "ppxlib"
+  "cppo"
```
Comment (dra27) [inline packages/genprint/genprint.0.2/opam]:
```suggestion
  "cppo" {build}
```

**Comment by progman1** [comment]
wrt the PR - how do I undo what I have done? I guess somehow retrieving the deleted file from my copy of your repo doesn't cut it?

**Comment by dra27** [comment]
Assuming you're at the root of your opam-repository checkout, `git checkout c1cb67a -- packages/genprint/genprint.0.1/opam` and commit the change.

If you don't mind my saying, that's a fundamental part of using Git - you might like to spend some time doing a few tutorials. GitHub has a relatively short [handbook](https://guides.github.com/introduction/git-handbook/). There's also [Pro Git](https://git-scm.com/book/en/v2) which is a text I personally recommend reading in its entirety.

**Comment by progman1** [comment]
I had hoped to get away with what I had acquired - well.. it was inadequate!
I will certainly look-up those suggestions soon. Thanks for the guidance.

**Comment by progman1** [comment]
this 'checking for ability to merge automatically' - it takes a long time.
what is it looking for and can I do anything to help it come to conclude in favour?

**Comment by kit-ty-kate** [comment]
Thanks!

**Comment by progman1** [comment]
will you consider my question?
eliminating the burden of this auto-check (which I think happened more than once
by avoiding it altogether has got to be a good thing, no?

**Comment by dra27** [comment]
@progman1 - it's checking whether your changes merge cleanly on to the `master` branch, essentially no conflicts in any files. For opam-repository, that's usually obvious, as PRs are just new files, but the delay is a GitHub issue - there's unlikely to be much you can do about it. That said, your PR branch was unusual because you added your new commits to your previous branch and resubmitted, rather than basing your commits freshly on `master` which *might* have caused some GitHub weirdness - it was certainly causing GitHub to display that your second PR was in fact a "first time contribution".

**Pull request merged**